packages/labs/src/select.tsx

```diff
@@ -0,0 +1,231 @@
+            {...props}
+            activeItem={this.state.activeItem}
+            onActiveItemChange={this.handleActiveItemChange}
+            onItemSelect={this.handleItemSelect}
+            query={this.state.query}
+            ref={this.refHandlers.inputList}
+            renderer={this.renderInputList}
+        />;
+    }
+
+    public componentDidUpdate(_prevProps: ISelectProps<T>, prevState: ISelectState<T>) {
+        if (this.state.isOpen && !prevState.isOpen && this.inputList != null) {
+            this.inputList.scrollActiveItemIntoView();
+        }
+    }
+
+    private renderInputList = (listProps: IInputListRendererProps<T>) => {
+        // not using defaultProps cuz they're hard to type with generics (can't use <T> on static members)
+        const { filterable = true, inputProps = {}, popoverProps = {} } = this.props;
+        const { handleKeyDown, handleKeyUp, query } = listProps;
+
+        const { ref, ...htmlInputProps } = inputProps;
+        const input = (
+            <InputGroup
+                autoFocus={true}
+                leftIconName="search"
+                onChange={this.handleQueryChange}
+                placeholder="Filter..."
+                rightElement={this.maybeRenderInputClearButton()}
+                value={query}
+                {...htmlInputProps}
+            />
+        );
+
+        return (
+            <Popover
+                autoFocus={false}
+                enforceFocus={false}
+                isOpen={this.state.isOpen}
+                position={Position.BOTTOM_LEFT}
+                {...popoverProps}
+                className={classNames(listProps.className, popoverProps.className)}
+                onInteraction={this.handlePopoverInteraction}
+                popoverClassName={classNames("pt-select-popover", popoverProps.popoverClassName)}
+                popoverWillOpen={this.handlePopoverWillOpen}
+                popoverDidOpen={this.handlePopoverDidOpen}
+                popoverWillClose={this.handlePopoverWillClose}
+            >
+                <div
+                    onKeyDown={this.state.isOpen ? handleKeyDown : this.handleTargetKeyDown}
+                    onKeyUp={this.state.isOpen && handleKeyUp}
+                >
+                    {this.props.children}
+                </div>
+                <div onKeyDown={handleKeyDown} onKeyUp={handleKeyUp}>
+                    {filterable ? input : undefined}
+                    <ul className={Classes.MENU} ref={listProps.itemsParentRef}>
+                        {this.renderItems(listProps)}
+                    </ul>
+                </div>
+            </Popover>
+        );
+    }
+
+    private renderItems({ activeItem, filteredItems, handleItemSelect }: IInputListRendererProps<T>) {
+        const { itemRenderer, noResults } = this.props;
+        return filteredItems.length > 0
+            ? filteredItems.map((item) => itemRenderer(item, item === activeItem, (e) => handleItemSelect(item, e)))
```

Comment:
Is it up to the consumer to add a ` key` to each item? Should we mention that in the docs?